Kudu.Core/Jobs/TriggeredJobSchedule.cs

```diff
@@ -0,0 +1,83 @@
+using System;
+using System.Linq.Expressions;
+using System.Threading;
+using Kudu.Contracts.Jobs;
+using Kudu.Core.Tracing;
+
+namespace Kudu.Core.Jobs
+{
+    public class TriggeredJobSchedule : IDisposable
+    {
+        private static readonly TimeSpan MaximumTimeSpanInterval = TimeSpan.FromDays(40);
+
+        private readonly Action<TriggeredJobSchedule> _onSchedule;
+
+        private Timer _timer;
+
+        public TriggeredJobSchedule(TriggeredJob triggeredJob, Action<TriggeredJobSchedule> onSchedule, TriggeredJobSchedulerLogger logger)
+        {
+            TriggeredJob = triggeredJob;
+            _onSchedule = onSchedule;
+            Logger = logger;
+
+            _timer = new Timer(OnTimer, triggeredJob, Timeout.Infinite, Timeout.Infinite);
+        }
+
+        public TriggeredJob TriggeredJob { get; private set; }
+
+        public Schedule Schedule { get; private set; }
+
+        public TriggeredJobSchedulerLogger Logger { get; private set; }
+
+        public void Reschedule(DateTime lastRun, Schedule schedule = null)
+        {
+            Schedule = schedule ?? Schedule;
+
+            var nextInterval = Schedule.GetNextInterval(lastRun);
+
+            // Limit next interval to 40 days so it won't conflict with the maximum time allowed to set a timer.
```

Comment:
This is handled here, after 40 days the timer will fire but since the schedule is not up yet it will reschedule again.